zulip_bots/zulip_bots/bots/github_detail/test_github_detail.py

```diff
@@ -89,3 +89,17 @@ def test_too_many_request(self) -> None:
 
         with self.mock_config_info(self.mock_config):
             self.verify_reply(request, bot_response)
+
+    def test_owner_and_repo_not_specified(self) -> None:
+        mock_invalid_config = {'owner': '', 'repo': ''}
+        invalid_request = '/#5365'
+        valid_request = 'zulip/zulip#5365'
+        bot_response = 'Failed to detect owner and repository name.'
+
+        with self.mock_config_info(mock_invalid_config):
+            self.verify_reply(invalid_request, bot_response)
+            self.verify_reply(valid_request, self.bot_github_response)
+
+        with self.mock_config_info(self.mock_config):
+            self.verify_reply(invalid_request, self.bot_github_response)
+            self.verify_reply(valid_request, self.bot_github_response)
```

Comment:
The tests now seem complete, but the different sub-tests in this function could be hard to follow for someone who sees this code for the first time.  Another thing is that every test here should use mocks for http traffic, instead of relying on network access. So to address both of these things, how about this:
* instead of reusing `bot_response_github`, which obfuscates the code, make new fixtures for the cases "valid request" and "invalid request" (for valid request, you can probably reuse an existing fixture, but you should still copy the response into the test functions).
* split the tests in this function into three functions: 
  * test_owner_and_repo_not_specified
  * test_owner_and_repo_specified_in_config_file
  * test_owner_and_repo_specified_in_message

Reply:
Thanks for your detailed review, I will do the changes. :)